aten/src/ATen/function_wrapper.py

```diff
@@ -878,8 +975,11 @@ def find_multidispatch_tensors(formals):
         # and Tensor arguments, you MUST only ever register it generically.
         r = []
         for formal in formals:
-            if formal['dynamic_type'] in ['TensorOptions', 'TensorList'] or is_any_tensor_type(formal):
+            if formal['dynamic_type'] in ['TensorList'] or is_any_tensor_type(formal):
                 r.append(formal['name'])
+
+        if check_if_factory_method(formals):
+            r.append('dtype, device, layout, pin_memory')
```

Comment:
This line feels counter to the goals of the refactor, since it is introducing a new site you have to update if you add a new argument to TensorOptions.